src/skfolio/distribution/multivariate/_vine_copula.py

```diff
@@ -0,0 +1,1145 @@
+                        u_max = 1.0 if np.isposinf(max_value) else dist.cdf(max_value)
+                    else:
+                        u_min, u_max = min_value, max_value
+
+                    # Sample uniformly in the transformed interval.
+                    samples = rng.uniform(low=u_min, high=u_max, size=n_samples)
+
+                elif np.isscalar(value):
+                    initial_cond[var] = value
+                    if self.log_transform:
+                        value = np.log(1 + value)
+                    if self.fit_marginals:
+                        # Transform conditioning value using the fitted marginal CDF.
+                        value = self.marginal_distributions_[var].cdf(value)
+                    samples = np.full(n_samples, value)
+                else:
+                    samples = np.asarray(value)
+                    initial_cond[var] = samples
+                    if samples.ndim != 1 or samples.shape[0] != n_samples:
+                        raise ValueError(
+                            "When an array is provided in `conditioning`, it must be a "
+                            f"1D array of length n_samples={n_samples}, got {samples.ndim}D of "
+                            f"length {samples.shape[0]}"
+                        )
+                    if self.log_transform:
+                        samples = np.log(1 + samples)
+                    if self.fit_marginals:
+                        # Transform conditioning samples using the fitted marginal CDF.
+                        samples = self.marginal_distributions_[var].cdf(samples)
+                uniform_cond_samples[var] = samples
+
+        # Determine sampling order based on vine structure.
+        sampling_order = self._sampling_order(conditioning_vars=conditioning_vars)
+
+        # Generate independent Uniform(0,1) samples for non-conditioned nodes.
+        X_rand = iter(rng.random(size=(n_samples, n_assets - len(conditioning_vars))).T)
+
+        # Initialize samples for each node according to the sampling order.
+        queue: deque[tuple[Node, bool]] = deque()
+        for node, is_left in sampling_order:
+            node_var = node.get_var(is_left)
+            if node_var in conditioning_vars:
+                init_samples = uniform_cond_samples[node_var]
+            else:
+                init_samples = next(X_rand)
+
+            # Avoid Inf
+            init_samples = np.clip(
+                init_samples,
+                a_min=_UNIFORM_SAMPLE_EPSILON,
+                a_max=1 - _UNIFORM_SAMPLE_EPSILON,
+            )
+
+            # For the root node, is_left is None.
+            if is_left is None:
+                node.u = init_samples
+            else:
+                queue.append((node, is_left))
+                if is_left:
+                    node.u = init_samples
+                else:
+                    node.v = init_samples
+
+        # Propagate samples through the vine structure.
```

Comment:
```suggestion
        # Propagate samples through the vine structure bottom-up.
```
Or a little bit more to help the reader